File: php/features.php
<?php
declare(strict_types=1);

// Unshortenme SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class UnshortenmeFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new UnshortenmeBaseFeature();
            case "test":
                return new UnshortenmeTestFeature();
            default:
                return new UnshortenmeBaseFeature();
        }
    }
}

File: py/features.py
# Unshortenme SDK feature factory

from feature.base_feature import UnshortenmeBaseFeature
from feature.test_feature import UnshortenmeTestFeature


def _make_feature(name):
    features = {
        "base": lambda: UnshortenmeBaseFeature(),
        "test": lambda: UnshortenmeTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
